src/display/window_file_list.cc

```diff
@@ -105,11 +105,15 @@ wstring_width(const std::string& i_str, int width) {
 
   // Pad with spaces to given width.
   while (swidth < width && length <= (unsigned int)width) {
-    result[length++] = ' ';
+    buf[length++] = ' ';
     swidth++;
   }
 
-  result[length] = 0;
+  buf[length] = 0;
+  
+  std::wstring result(buf);
+
+  free(buf);
```

Comment:
@stickz On a cursory view, this makes no sense to me. Where do you see a memory leak? Moreover, `buf` is allocated on the  stack, freeing it is an error.